### PR TITLE
[WFLY-17740] Add missing @fallback test in Microprofile Fault Tolerance quickstart

### PR DESCRIPTION
**Pull request by ChristinaDsl**
https://issues.redhat.com/browse/WFLY-17740
In this issue were added two Test methods, both testing a method with @Fallback annotation . The first one tests the successful execution of the method, which also has specific time frame (with the help of @Timeout). The second one secures that every time the execution time of the method exceeds the time limit, the Fallback method executed successfully.

### CODE REVIEW

**Comment by iweiss** [comment]
Please add the link to the issue tracker on the first line

**Comment by emmartins** [comment]
@pferraro @rhusar not sure who is now the responsible for this QS, my guess it's one of you, please review

**Comment by rhusar** [comment]
@emmartins I can't assign myself as reviewer - can you please fix the project permissions settings? Thanks!

File: microprofile-fault-tolerance/src/main/java/org/wildfly/quickstarts/microprofile/faulttolerance/CoffeeResource.java [hunk 1]
```diff
@@ -166,9 +170,17 @@ private void maybeFail(String failureLogMessage) {
         }
     }
 
-    private void randomDelay() throws InterruptedException {
+    private void randomDelay() throws InterruptedException  {
```
Comment (rhusar) [inline microprofile-fault-tolerance/src/main/java/org/wildfly/quickstarts/microprofile/faulttolerance/CoffeeResource.java:173]: The extra new space here is wrong.

File: microprofile-fault-tolerance/src/test/java/org/wildfly/quickstarts/microprofile/faulttolerance/MicroProfileFaultToleranceITest.java [hunk 2]
```diff
@@ -33,10 +33,11 @@
 import jakarta.ws.rs.core.GenericType;
 import jakarta.ws.rs.core.Response;
 import java.net.URL;
+import java.util.ArrayList;
 import java.util.List;
 
 @RunWith(Arquillian.class)
-public class MicroProfileFaultToleranceIT {
+public class MicroProfileFaultToleranceITest {
```
Comment (rhusar) [inline microprofile-fault-tolerance/src/test/java/org/wildfly/quickstarts/microprofile/faulttolerance/MicroProfileFaultToleranceITest.java:40]: The rename here is unnecessary and would make this out of sync with other quickstarts in this category.

File: microprofile-fault-tolerance/src/main/java/org/wildfly/quickstarts/microprofile/faulttolerance/CoffeeResource.java [hunk 3]
```diff
@@ -131,7 +135,7 @@ public Response availability(@PathParam("id") int id) {
 
     @GET
     @Path("/{id}/recommendations")
-    @Timeout(250)
+    @Timeout(value = 250)
```
Comment (rhusar) [inline microprofile-fault-tolerance/src/main/java/org/wildfly/quickstarts/microprofile/faulttolerance/CoffeeResource.java:138]: This change is unnecessary and also inconsistent with e.g. the annotation above it. The `value =` is implicit

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by ChristinaDsl** [comment]
@rhusar Thank you. All changes have been made.

**Comment by ChristinaDsl** [comment]
@rhusar Is there anything else that you need from my side?

**Comment by rhusar** [comment]
> @rhusar Is there anything else that you need from my side?

Nope - just competing priorities on my side...

**Review by emmartins — CHANGES_REQUESTED** [review]
The quickstart testing has been fully reworked, please rebase and update changes.

File: microprofile-fault-tolerance/pom.xml [hunk 4]
```diff
@@ -167,6 +167,11 @@
             <artifactId>junit</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.wildfly.arquillian</groupId>
+            <artifactId>wildfly-arquillian-container-managed</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (rhusar) [inline microprofile-fault-tolerance/pom.xml:174]: Remove this as this won't work when testing against deployment running in e.g. OpenShift.

File: microprofile-fault-tolerance/pom.xml [hunk 5]
```diff
@@ -200,6 +205,12 @@
             <artifactId>resteasy-jackson2-provider</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.arquillian</groupId>
+            <artifactId>arquillian-api</artifactId>
+            <version>1.0.0.Alpha5</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (rhusar) [inline microprofile-fault-tolerance/pom.xml:213]: Remove this as this won't work when testing against deployment running in e.g. OpenShift.

File: microprofile-fault-tolerance/src/test/resources/arquillian.xml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+         If you prefer not to define the JBOSS_HOME environment variable, alternatively you can uncomment the
+         following `jbossHome` property and replace ${jboss.home.name} with the path to your ${product.name} installation. -->
+
+      <!-- In order to run tests, commend out "jbossHome" line -->
+      <configuration>
+          <property name="serverConfig">standalone-microprofile.xml</property>
+          <!--<property name="jbossHome">${jboss.home.name}</property>-->
+      </configuration>
+  </container>
+</arquillian>
```
Comment (rhusar) [inline microprofile-fault-tolerance/src/test/resources/arquillian.xml:39]: We no longer use ARQ. Remove this as this won't work when testing against deployment running in e.g. OpenShift.

File: microprofile-fault-tolerance/src/test/java/org/wildfly/quickstarts/microprofile/faulttolerance/CoffeeResourceIT.java [hunk 7]
```diff
@@ -17,28 +17,47 @@
...
+    private static final String APP_NAME = "microprofile-fault-tolerance";
+
+    @ArquillianResource
+    private URL deploymentUrl;
+
+    @Inject
+    private CoffeeResource coffeeResource;
+
+    @Inject
+    CoffeeRepositoryService coffeeRepository;
```
Comment (rhusar) [inline microprofile-fault-tolerance/src/test/java/org/wildfly/quickstarts/microprofile/faulttolerance/CoffeeResourceIT.java:59]: Remove this as this won't work when testing against deployment running in e.g. OpenShift.

**Review by rhusar — CHANGES_REQUESTED** [review]
@ChristinaDsl Please review the prior changes to the testing and note that we have removed ARQ and reworked to use a remote client istead.

**Comment by ChristinaDsl** [comment]
@rhusar thanks for your comments. I will manage the fixes.

**Comment by emmartins** [comment]
@ChristinaDsl please verify the new test, one run on CI did fail

**Comment by rhusar** [comment]
@emmartins Yet again - could you please verify the permissions for this repository? I can't even mark (my own) conversations as resolved...

**Comment by emmartins** [comment]
@rhusar if I find that option... yes sure!!!!

**Comment by rhusar** [comment]
> @rhusar if I find that option... yes sure!!!!

I am going to guess that it's just the triage permission.

**Comment by emmartins** [comment]
can you please recheck now?

**Comment by ChristinaDsl** [comment]
@emmartins  Sure.

**Comment by emmartins** [comment]
Merging since testsuite seems stable now, if an intermitent pops up again will open an issue, thanks @ChristinaDsl

**Comment by rhusar** [comment]
Thanks @ChristinaDsl  and @emmartins !

**Comment by ChristinaDsl** [comment]
Thank you too @emmartins and @rhusar!

**Pull request merged**